cli/lsp/language_server.rs

```diff
@@ -139,9 +139,6 @@ pub struct LanguageServer {
   init_flag: AsyncFlag,
```

Comment:
@nayeemrmn do you think we could move the init_flag and shutdown_flag into our tower-lsp fork? Might help simplify things up here.

Reply:
`shutdown_flag` is a workaround for a tower lsp bug, so yeah we should just fix that #20700.

https://github.com/denoland/deno/blob/d29f9f99dd51ea1cac5cf28dc1444e0f416e6957/cli/lsp/mod.rs#L79-L90

`init_flag` I realised yesterday is buggy, if you type up a storm before it resolves the notification handlers will unblock out of order. So we might need to move that too with a proper queue implementation?